packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2272,8 +2272,13 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
 
   @override
   void didChangeMetrics() {
-    if (_lastBottomViewInset < WidgetsBinding.instance.window.viewInsets.bottom) {
-      _scheduleShowCaretOnScreen();
+    if (_lastBottomViewInset != WidgetsBinding.instance.window.viewInsets.bottom) {
+      SchedulerBinding.instance.addPostFrameCallback((Duration _) {
```

Comment:
Out of curiosity, why does this need to be in a post-frame callback? `didChangeMetrics` is not going to be called during `drawFrame` so it sounds like it's OK to just call `updateForScroll` here?

Reply:
It seems that at the time the time that the toolbar gets rebuilt without a postframecallback, the `renderObject` hasn't been updated to the new position (see [here](https://github.com/flutter/flutter/blob/0d7e6798c9a4677f347354c3571e230e4179bfe9/packages/flutter/lib/src/widgets/text_selection.dart#L620-L623)).

Actually I just tried removing it and it only broke the case of closing the keyboard, not opening it.  I thought it was necessary for both when I wrote this but maybe not.

Reply:
Ah makes sense. Thanks for the explanation!